internal/lib/sandbox/builder.go

```diff
@@ -123,37 +123,140 @@ type Builder interface {
+	config      *types.PodSandboxConfig
+	infra       container.Container
+	sandboxRef  *Sandbox
+	validations sandboxValidations
 }
 
 // NewBuilder creates a new, empty Sandbox instance.
 func NewBuilder() Builder {
 	return &sandboxBuilder{
-		config:     nil,
-		sandboxRef: new(Sandbox),
+		config:      nil,
+		sandboxRef:  new(Sandbox),
+		validations: 0,
 	}
 }
 
 // GetSandbox gets the sandbox and deletes the config and sandbox.
 // TODO: Add validations before returning the sandbox.
 func (b *sandboxBuilder) GetSandbox() (*Sandbox, error) {
-	if b.sandboxRef.criSandbox == nil {
-		return nil, errors.New("cri-o sandbox not initialized")
+	if err := b.Validate(); err != nil {
+		return nil, err
 	}
 
 	sandboxRef := b.sandboxRef
 	b.config = nil
 	b.sandboxRef = nil
+	b.validations = 0
 
 	return sandboxRef, nil
 }
 
+// allValidations represents all required validation flags combined.
+const allValidations sandboxValidations = validationSetCRISandbox |
+	validationSetCreatedAt |
+	validationSetID |
+	validationSetName |
+	validationSetLogDir |
+	validationSetShmPath |
+	validationSetNamespace |
+	validationSetKubeName |
+	validationSetProcessLabel |
+	validationSetMountLabel |
+	validationSetCgroupParent |
+	validationSetPrivileged |
+	validationSetRuntimeHandler |
+	validationSetResolvPath |
+	validationSetHostname |
+	validationSetPortMappings |
+	validationSetHostNetwork |
+	validationSetUsernsMode |
+	validationSetPodLinuxOverhead |
+	validationSetPodLinuxResources
+
+// Validate validates the sandbox.
+func (b *sandboxBuilder) Validate() error {
+	// Quick check if all validations are set
+	if b.validations == allValidations {
+		return nil
+	}
+
+	// If not all validations are set, check which ones are missing.
+	validations := map[string]sandboxValidations{
+		"cri-o sandbox":                validationSetCRISandbox,
+		"createdAt time":               validationSetCreatedAt,
+		"sandbox id":                   validationSetID,
+		"sandbox name":                 validationSetName,
+		"sandbox logDir":               validationSetLogDir,
+		"sandbox shmPath":              validationSetShmPath,
+		"sandbox setNamespace":         validationSetNamespace,
+		"sandbox setKubeName":          validationSetKubeName,
+		"sandbox setProcessLabel":      validationSetProcessLabel,
+		"sandbox setMountLabel":        validationSetMountLabel,
+		"sandbox setCrgroupParent":     validationSetCgroupParent,
+		"sandbox setPrivileged":        validationSetPrivileged,
+		"sandbox setRuntimeHandler":    validationSetRuntimeHandler,
+		"sandbox setReslovPath":        validationSetResolvPath,
+		"sandbox setHostname":          validationSetHostname,
+		"sandbox setPortMappings":      validationSetPortMappings,
+		"sandbox setHostNetwork":       validationSetHostNetwork,
+		"sandbox setUsernsMode":        validationSetUsernsMode,
+		"sandbox setPodLinuxOverhead":  validationSetPodLinuxOverhead,
+		"sandbox setPodLinuxResources": validationSetPodLinuxResources,
+	}
```

Comment:
We can make these keys consistent, for example to use field name after `validationSet*`.

And we can use package level var instead of function level to avoid allocating this map in every call (or is it optimised by compiler?).